.github/Dockerfile-crux-mir

```diff
@@ -2,6 +2,8 @@ FROM rust:1.46.0 AS mir_json
 
 ADD dependencies/mir-json /mir-json
 WORKDIR /mir-json
+# Work around https://github.com/rust-lang/cargo/issues/10303
+ENV CARGO_NET_GIT_FETCH_WITH_CLI=true
```

Comment:
Just double checking: this https://github.com/rust-lang/cargo/issues/10303#issuecomment-1241342524 makes it sound like the issue was resolved and this is no longer needed?

Reply:
It has been fixed in more recent Rust/`cargo` distributions, but we are still using an old Rust nightly that is affected by this issue:

https://github.com/GaloisInc/crucible/blob/5ae12b7598a578b7dd7c42671113572c218a0f0f/.github/workflows/crux-mir-build.yml#L89

(I am working to upgrade the Rust nightly to a 2023 one, but it's not going to be ready in time for the Crux 0.7 release.)